index.js

```diff
@@ -434,20 +446,32 @@ function main (options) {
           stack.push(value)
           let i = 0
           for (; i < value.length - 1; i++) {
-            const tmp = stringifySimple(i, value[i], stack)
+            const tmp = stringifySimple(i, value[i], stack, depth)
             res += tmp !== undefined ? tmp : 'null'
             res += ','
           }
-          const tmp = stringifySimple(i, value[i], stack)
+          const tmp = stringifySimple(i, value[i], stack, depth)
           res += tmp !== undefined ? tmp : 'null'
           stack.pop()
           return `[${res}]`
         }
 
         let keys = Object.keys(value)
+        if (maximumBreadth) {
+          const removedKeys = keys.length - maximumBreadth
+          if (removedKeys > 0) {
+            keys = keys.slice(0, maximumBreadth)
+            keys.push('[DEBUG]')
+            value['[DEBUG]'] = `${removedKeys} keys not stringified`
+          }
+        }
```

Comment:
Thanks for implementing it. :-) I am not yet 100% certain if [DEBUG] is an ideal property name but I can't think of a better one right now.

```suggestion
```

The current implementation would not work properly for typed arrays. Those use a fast path that does skip the regular keys. We have to pass the breadth option to the `stringifyTypedArray()` function and switch this line

https://github.com/BridgeAR/safe-stable-stringify/pull/20/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R484

to 

```js
const maximum = Math.min(keys.length, options.maximumBreadth)
for (let index = 0; index < maximum; index++) {
  key = keys[index]
```

Reply:
why don't use a symbol instead?

Reply:
@simoneb I am not certain I can follow. How should that be represented? Symbols are ignored during stringification. They have no representation in JSON and IMO the implementation should adhere to JSON to be readable by any JSON parse implementation.

Reply:
I was talking about representing the information we remove. Clearly if we put it into a property which is also JSON-stringified we're back to square one and the process will hang

Reply:
at the moment I am not checking arrays, but only objects, should we extend this feature to arrays also?

Reply:
Yes please